core/include/detray/definitions/track_parameterization.hpp

```diff
@@ -9,7 +9,7 @@ namespace detray {
 
 /// Components of a bound track parameters vector.
 ///
-enum bound_indices : unsigned int {
+enum bound_indices : __plugin::size_type {
```

Comment:
Maybe you are talking about SMatrix which uses `unsigned int`? Either way, `__plugin::size_type` is only used for the array or matrix size type. thus, we don't have to consider negative value cases

Reply:
Actually I meant  
```/// size type for Eigen storage model
using size_type = int;```
in `storage/eigen/include/algebra/storage/eigen.hpp`, which theoretically allows those index types to be negative now. But I guess, as long as we don't cast anything into e.g. ```bound_indices``` we should be fine.

Reply:
Ah I see the point. Yeah it should be fine I hope

Reply:
If we do arithmetic anywhere to calculate an index, then cast and then use the operator ```[]``` to fetch the value, instead of e.g. ```vector.at()``` or ```detail::get<>```, we might get some nasty errors, no?

Reply:
True but changing `__plugin::size_type` into `unsigned int` doesn't solve the problem

Reply:
Hm, apparently you are right: Its weird behavior/undefined behavior (```unsigned int```) vs just undefined behavior (```int```). The gist seems to be then, don't use ```[]``` unless one knows what one is doing :)